File: c++/alturas.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {

    int N, cont = 0, porcentagem;
    double media, somaAltura = 0;

    cout << "Quantas pessoas serao digitadas? ";
    cin >> N;

    string nome[N];
    int idade[N];
    double altura[N];

    for(int i = 0; i < N; i++) {
        cout << "Dados da " << i+1 << "a pessoa: " << endl;
        cout << "Nome: ";
        cin.ignore(INT_MAX, '\n');
        getline(cin, nome[i]);
        cout << "Idade: ";
        cin >> idade[i];
        cout << "Altura: ";
        cin >> altura[i];
    }

    for(int i = 0; i < N; i++) {
        somaAltura = somaAltura + altura[i];
    }
    media = somaAltura / N;

    cout << fixed << setprecision(2);
    cout << endl << "Altura media: " << media << endl;

    for(int i = 0; i < N; i++) {
        if(idade[i] < 16) {
            cont = cont + 1;
        }
    }

    porcentagem = cont * 100.0 / N;

    cout << fixed << setprecision(1);
    cout << "Pessoas com menos de 16 anos: " << porcentagem << "%" << endl;

    for(int i = 0; i < N; i++) {
        if(idade[i] < 16) {
            cout << nome[i] << endl;
        }
    }

    return 0;
}

File: java/alturas.java
import java.util.Locale;
import java.util.Scanner;

public class alturas {

	public static void main(String[] args) {
		
		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);
		
		int N, cont;
		double somaAltura, mediaAltura, porcentagem;
		
		System.out.print("Quantas pessoas serao digitadas? ");
		N = sc.nextInt();
		
		String[] vetNome = new String[N];
		int[] vetIdade = new int[N];
		double[] vetAltura = new double[N];
		
		for (int i = 0; i < N; i++) {
			System.out.println("Dados da " + String.format("%d", i + 1) + "a pessoa: ");
			System.out.print("Nome: ");
			vetNome[i] = sc.next();
			System.out.print("Idade: ");
			vetIdade[i] = sc.nextInt();
			System.out.print("Altura: ");
			vetAltura[i] = sc.nextDouble();
		}
		System.out.println();
		
		somaAltura = 0;
		for (int i = 0; i < N; i++) {
			somaAltura = somaAltura + vetAltura[i];			
		}
		
		mediaAltura = somaAltura / N;
		System.out.println("Altura média: " + String.format("%.2f", mediaAltura));
		
		cont = 0;
		for (int i = 0; i < N; i++) {
			if (vetIdade[i] < 16) {
				cont = cont + 1;
			}
		}
		
		porcentagem = ((double)cont / N) * 100.0;
		System.out.println("Pessoas com menos de 16 anos: " + 
				   String.format("%.1f", porcentagem) + "%");
		
		for (int i = 0; i < N; i++) {
			if (vetIdade[i] < 16) {
				System.out.println(vetNome[i]);
			}
		}
		
		sc.close();
	}

}
